ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/TableSelectBuilder.java

```diff
@@ -26,18 +29,28 @@
 import io.confluent.ksql.execution.transform.select.Selection;
 import io.confluent.ksql.logging.processing.ProcessingLogger;
 import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import java.util.Optional;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
 import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.state.KeyValueStore;
 
 public final class TableSelectBuilder {
+  private static final String PROJECT_OP = "Last";
 
   private TableSelectBuilder() {
   }
 
   @SuppressWarnings("unchecked")
   public static <K> KTableHolder<K> build(
-      final KTableHolder<K> table,
-      final TableSelect<K> step,
-      final RuntimeBuildContext buildContext
+          final KTableHolder<K> table,
+          final TableSelect<K> step,
+          final RuntimeBuildContext buildContext,
+          final Formats formats,
```

Comment:
Should formats be Optional?  It will only exist if the plan comes from a new version of kql, right?

Reply:
I have made it Optional for the constructor that is invoked by Jackson

Reply:
I will try

Reply:
done